terraform/application/storage_account.tf

```diff
@@ -0,0 +1,37 @@
+resource "azurerm_storage_account" "general_storage_account" {
+  count                            = var.create_storage_account ? 1 : 0
```

Comment:
you could replace this with 
count  = var.storage_account_name != null ? 1 : 0
and then drop the create_storage_account variable.
But as it looks like you also create the storage account for every env, you could just drop the count and var completely?